core/trino-main/src/main/java/io/trino/json/JsonInputErrorNode.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+import java.io.IOException;
+import java.util.List;
+
+public class JsonInputErrorNode
+        extends JsonNode
+{
+    public static final JsonInputErrorNode JSON_ERROR = new JsonInputErrorNode();
+
+    private JsonInputErrorNode() {}
+
+    @Override
+    public <T extends JsonNode> T deepCopy()
+    {
+        throw new UnsupportedOperationException();
```

Comment:
technically you can return `this` here as it has no fields.

Reply:
Yeah, but in this case I'd rather not implement methods I don't use. This node is just internal stuff. If someone ever happens to call `deepCopy()` on it, they'd better stop here and have the chance to reconsider.